src/communication/hpxreceiver.h

```diff
@@ -37,9 +42,17 @@ class HPXReceiver : public hpx::components::simple_component_base<HPXReceiver<CA
 
     static hpx::future<boost::shared_ptr<HPXReceiver> > make(const std::string& name, std::size_t rank = 0)
     {
```

Comment:
Conceptual question: why do you create a component if all you are storing is the `shared_ptr<>` referencing it? Why not directly creating the object in this case?